### ISSUE TITLE
CMake configuration for library versions/flavors (avoid DLL collisions).

### ISSUE THREAD

**Issue by pmuron** [F2I-Consulting/fesapi#206]
**Is your feature request related to a problem? Please describe.**
FESAPI is being used by multiple vendors and groups, delivering as plug-in to a common windows-based platform, residing in same process.
We have a number of issues because as many parties use the same DLL name (e.g. FesapiCpp.dll or FesapiCpp.1.0.0.dll) for what appears plain different or slightly mod-ed version. 

**Describe the solution you'd like**
CMake should include a simple way to generate a "flavor" postfix (like boost does) to separate and identify (and load) each DLLs (e.g. I could have a FesapiCpp-CVX-1.0.0.dll).

**Describe alternatives you've considered**
Use of Fesapi_VERSION_TWEAK kind-of works but requires some hand-editing of CS AssemblyInfo because it parts from expected semantic versioning scheme. Again not a super big deal, little things.

**Additional context**

**Comment by philippeVerney** [comment]
It looks ok with me.
No deadline, no priority but indeed on the todo list.

As a side note, it is a bit (symbol conflict, not dll name conflict) related with #28

**Comment by philippeVerney** [comment]
Hi Pierre,

I think I have found the simple way to generate a "flavor" postfix which CMake includes : https://cmake.org/cmake/help/v3.16/variable/CMAKE_CONFIG_POSTFIX.html

So, just set a CMAKE_RELEASE_POSTFIX (or CMAKE_DEBUG_POSTFIX) cmake variable to the postfix you want, configure and generate and everything should be fine.
FYI, I have tested with CMAKE_RELEASE_POSTFIX=-CVX in cmake-gui by clicking on the "Add Entry" button and it has generated a "FesapiCpp-CVX.1.1.0.0.dll" library.

FYI, this is what I have used to solve #205 by automatically providing a default value of "d" to CMAKE_DEBUG_POSTFIX.

I think I just have to propagate this postfix to the C# dll as well now...
